### ISSUE TITLE
FilesystemServiceProvider -  Allowed memory size exhausted

### ISSUE THREAD

**Issue by tdicks** [laravel/lumen-framework#130]
I'm using Lumen as a gateway for files hosted on DreamObjects. I've created a facade named `CloudStorage` which accesses the `filesystem.cloud` singleton defined in `Illuminate\Filesystem\FilesystemServiceProvider`. This gives me an easy facade to use for working with the default cloud storage filesystem, whilst retaining `Storage` for any local storage operations.

The facade is in app/Facades/CloudStorage.php and is namespaced properly. It works fine in Laravel 5 but using Lumen throws out an allowed memory size exhausted error from PHP . The full error is:

**Fatal error**: Allowed memory size of 94371840 bytes exhausted (tried to allocate 130968 bytes) in **<app_path>/vendor/illuminate/container/Container.php** on line **366**

`$app->withFacades();` is present in `bootstrap/app.php`, as is a `class_alias()` call for the facade itself. I've also added `$app->register('Illuminate\Filesystem\FilesystemServiceProvider');` to register the service provider. If I comment this line, I get a different error, this time from Lumen: `Class filesystem.cloud does not exist`.

The second error makes sense as the FilesystemServiceProvider obviously isn't registered. However when it is registered, the allowed memory error appears.

I could work around this by using `Storage::disk('dreamobjects_fs')` for each DO operation, but I'd rather keep things simple with just a single facade.

Any ideas on this?

**Comment by tdicks** [comment]
I appear to have fixed my own problem. If I understand it correctly, I was causing an infinite loop by continuously loading the FilesystemServiceProvider.

I checked what was happening in Lumen's Application class and noticed it loads the filesystem singleton in the `registerFilesBindings()` method. I checked FSP again (`registerFlysystem()`) and basically copied the logic there in my bootstrap/app.php:

```php
$app->singleton(
        'filesystem.cloud',
        function () use ($app)
        {
                return $app['filesystem']->disk( $app['config']['filesystems.cloud'] );
        }
);
```
This seems to be working fine although I would appreciate any comments on whether these findings are correct or not.
